### DISCUSSION TITLE
Sendjng of ping

### DISCUSSION THREAD

**Discussion by Notadumdum** [miguelgrinberg/python-socketio#1021] [Q&A] [answered]
Hey i have a question.. does this library automatically send pings..
I am trying to create a socket.io connection to a server and i want to know if this kind of ping is sent automatically
![20220912_182106](https://user-images.githubusercontent.com/105162586/189717190-cf1e7b83-409c-4273-9b04-c69179254475.jpg)

**Comment by miguelgrinberg** [comment, marked as answer]
Are you asking about the client functionality in this library or the server?

Your screenshot shows the browser sending to the server, in case it isn't clear.

The "2" packets are ping packets, sent at regular intervals and automatically. This is an old version of the Socket.IO protocol, in current versions the server sends pings to clients.

The "42" packet is an application specific packet.

**Comment by Notadumdum** [comment]
Yes i am aware the browser sends 2 to the server.. i am trying to reproduce this in vscode... so since you said this is an old version of socket.io.. how will i make use of it.. in case you are a bit confused I asked a question on connecting some time back.. so is there any difference in connecting using the new version or the old version

https://github.com/miguelgrinberg/python-socketio/discussions/991

**Reply by miguelgrinberg** [reply]
Still confused. What does vscode have to do with this? Also you did not answer my first question. Do you want to build a client or a server with this library?

If you are trying to connect with some other client or server that uses an old version of the Socket.IO protocol, then you need to switch to an old version of this library that matches what you need, probably a 4.x version instead of the current 5.x line.

**Reply by Notadumdum** [reply]
I want to build a client with this library.. it is vscode i am using to write the code.that is why i mentioned it in the first place.... Also is the documentation different for each versions?
What is your advice?
Thanks in advance

**Reply by miguelgrinberg** [reply]
As I said above, try the latest 4.x version of this library for your client as that is probably the one that is compatible with the server that you are connecting to. Install the following versions:

```bash
pip install python-engineio==3.14.2
pip install python-socketio==4.6.1
```

The documentation and examples that correspond to this old version are [here](https://github.com/miguelgrinberg/python-socketio/tree/v4.6.1).

**Reply by Notadumdum** [reply]
Thank you brother.. is the documentation of the old version the same with the documentation of the new one.. acsue at the bottom it shows ```latest```

**Reply by miguelgrinberg** [reply]
No. The documentation comes with the source: https://github.com/miguelgrinberg/python-socketio/tree/v4.6.1/docs. Anything other than the latest version you need to build yourself.

**Reply by Notadumdum** [reply]
Thanks
